core/exception/Exception.ts

```diff
@@ -150,19 +161,27 @@ export class Exception {
             name: 'Server Unavailable',
             message,
             httpStatus: 0, // native fetch doesn't put status on its Error
-            originalMessage: e.message,
+            originalMessage: e?.message,
             fetchOptions,
             stack: null
-        });
+        }) as FetchException;
     }
 
     //-----------------------
     // Implementation
     //-----------------------
-    private static createInternal(defaults, override = {}) {
-        if (isString(override)) {
-            override = {message: override};
-        }
-        return Object.assign(new Error(), defaults, override, {isHoistException: true});
+    private static createInternal(
```

Comment:
Not sure if this is any better, but was thinking this could also be made generic to avoid the casting.

```
private static createInternal<E extends HoistException = HoistException>(
    defaults: PlainObject,
    override: PlainObject = {}
): E
```

Reply:
Still requires specifying `createInternal<FetchException>(...`.  Would be nice if the argument types could inform the return type, but maybe I'm overthinking this for an internal private method.

Reply:
Sure seems marginally better -- good idea